doc/design/publish-1.md

```diff
@@ -0,0 +1,141 @@
+# Moving Parts
+
+This is a summary of system components that are relevant to publish and how they relate to eachother.
+
+### Dandiset
+A DANDI dataset.
+Each dandiset is guaranteed to have a draft Version, and can also have any number of published Versions.
```

Comment:
clarification: ATM "is guaranteed to have a draft Version" is true.  In the future we did think about possibility to archive dandisets, so no mutable `draft`  would be left behind, only published Versions.

Reply:
Lets leave this for now, and clarify with a bigger, better design doc when we care about that feature.